core/domain/state_domain_test.py

```diff
@@ -598,6 +601,133 @@ def test_validate_state_solicit_answer_details(self):
         self.assertEqual(init_state.solicit_answer_details, True)
 
 
+class ImageDomainUnitTests(test_utils.GenericTestBase):
+    """Test methods operate on Image."""
+    def test_from_and_to_dict_works_correctly(self):
+        image_dict = {
+            'src': '',
+            'placeholder': True,
+            'instructions': 'no instructions'
+        }
+        image_object = state_domain.Image.from_dict(image_dict)
+        self.assertEqual(image_object.to_dict(), image_dict)
+
+class ImageAssetsDomainUnitTests(test_utils.GenericTestBase):
+    """Test methods operating on Image Assets."""
+    def test_from_and_to_dict_works_correctly(self):
+        image_assets_dict = {
+            'image_mapping': {
+                'image_id_1': {
+                    'src': '',
+                    'placeholder': True,
+                    'instructions': 'no instructions'
+                },
+                'image_id_2': {
+                    'src': '',
+                    'placeholder': False,
+                    'instructions': 'no instructions'
+                }
+            }
+        }
+
+        image_assets = state_domain.ImageAssets.from_dict(image_assets_dict)
+        self.assertEqual(image_assets.to_dict(), image_assets_dict)
+
+    def test_delete_image(self):
+        image_assets_dict = {
+            'image_mapping': {
+                'image_id_1': {
+                    'src': '',
+                    'placeholder': True,
+                    'instructions': 'no instructions'
+                },
+                'image_id_2': {
+                    'src': '',
+                    'placeholder': False,
+                    'instructions': 'no instructions'
+                }
+            }
+        }
+
+        expected_image_assets_dict = {
+            'image_mapping': {
+                'image_id_1': {
+                    'src': '',
+                    'placeholder': True,
+                    'instructions': 'no instructions'
+                }
+            }
+        }
+
+        image_assets = state_domain.ImageAssets.from_dict(image_assets_dict)
+        image_assets.delete_image('image_id_2')
```

Comment:
What's the expected behavior if image_id doesn't exist? Please add a test for this as well.